hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -266,17 +299,29 @@ public AbfsRestOperation createPath(final String path, final boolean isFile, fin
     return op;
   }
 
-  public AbfsRestOperation renamePath(final String source, final String destination, final String continuation)
+  public AbfsRestOperation renamePath(String source, final String destination, final String continuation)
           throws AzureBlobFileSystemException {
     final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
+    String sasToken = "";
+
+    if (authType == AuthType.SAS) {
+      final AbfsUriQueryBuilder queryBuilder = new AbfsUriQueryBuilder();
+      appendSASTokenToQuery(source, SASTokenProvider.RENAME_SOURCE_OPERATION, queryBuilder);
+      sasToken = queryBuilder.toString();
+    }
 
-    final String encodedRenameSource = urlEncode(FORWARD_SLASH + this.getFileSystem() + source);
+    final String encodedRenameSource =
+        urlEncode(FORWARD_SLASH + this.getFileSystem() + source) + sasToken;
     requestHeaders.add(new AbfsHttpHeader(X_MS_RENAME_SOURCE, encodedRenameSource));
     requestHeaders.add(new AbfsHttpHeader(IF_NONE_MATCH, STAR));
 
     final AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
     abfsUriQueryBuilder.addQuery(QUERY_PARAM_CONTINUATION, continuation);
 
+    if (authType == AuthType.SAS) {
```

Comment:
this is replicated enough it should be its own method, e.g

```
maybeAppendSASToken(path, operation, querybuilder)
 if (authType=SAS) {
   appendSASTokenToQuery(path, SASTokenProvider.RENAME_SOURCE_OPERATION, queryBuilder);
 }
```

and use wherever needed

Reply:
Moved the authType check to appendSASToken method, though I have not renamed the method that adds the SAS token. Looks fine the way it is with the check for authType done within.